app/services/user_signup.rb

```diff
@@ -0,0 +1,47 @@
+class UserSignup
+  attr_accessor :error_message
+
+  def initialize(user)
+    @user = user
+  end
+
+  def sign_up(stripe_token, invitation_token)
+    if @user.valid?
+      charge = StripeWrapper::Charge.create(
+        amount:      999,
+        source:      stripe_token,
+        description: "Sign up charge for #{@user.email}"
+      )
+      if charge.successful?
+        @user.save
+        handle_invitation(invitation_token)
+        AppMailer.delay.welcome_email(@user.id)
+        @status = :success
+        self
```

Comment:
All the paths lead to `self`. You can actually move this out of the individual paths and return self on the final line.